blog/_posts/2021-08-16-write-more-bash.md

```diff
@@ -0,0 +1,206 @@
+---
+slug: "write-more-bash"
+date: 2021-08-16
+author: nick
+layout: blog
+title: "Write More Bash to Hack Features Faster and with Less Testing"
+subtitle: "A walkthrough of the 'cancel' button extension"
+description: "A walkthrough of the 'cancel' button extension"
+image: "/assets/images/write-more-bash/bonk.jpg"
+image_type: "contain"
+tags:
+  - api
+  - bash
+---
+
+The KubeCon schedule was announced last week. The Tilt team is giving two talks
+😇.
+
+- ["Beyond Kubernetes Security"](https://sched.co/lV4f) - A "blockbuster action thriller."
+- ["The Control Loop as an Application Development Framework"](https://sched.co/lV1E) - How to write apps with more control loops.
+
+Both talks are tangentially related to Tilt at best! They're mostly about
+Kubernetes, the ideas behind Kubernetes, and how those ideas can (and should!)
+affect your life as a dev.
+
+In the Control Loop talk, I'm going to talk more about the Kubernetes ideas and
+libraries that Tilt is built on top of. The Kubernetes libraries have a lot of
+tools to help us build robust systems.  But even better, they also have a lot of
+tools that let you hack together shit fast in Bash.
+
+What does this look like when we're developing Tilt? I'm going to show you a
+Bash-based Tilt extension I wrote last week! And as we go through it, you'll learn:
+
+- Why Kubernetes stuff lends itself so well to Bash
+- What a Bash-based Tilt extension looks like
+- How you too can hack stuff into Tilt with Bash
+
+## Why Kubernetes stuff lends itself so well to Bash
+
+The core idea of Kubernetes is simple: if you want to manage servers, you need a way
+to react to changes at runtime.
+
+On top of that, Kubernetes offers a suite of data models, HTTP APIs, and CLIs
+that are really consistent and well thought-out! (See: ["Kubernetes Is So Simple
+You Can Explore it with
+Curl"](https://blog.tilt.dev/2021/03/18/kubernetes-is-so-simple.html).)
+
+This API platform allows teams to create very complex, robust features.
+But few people appreciate that you can also use it to build simple, hacky features!
+
+A few weeks ago [L](https://twitter.com/ellenkorbes) asked if we could add a
+cancel button to the Tilt UI to kill a running local server. And I realized that you could
+hack this together with a couple dozen lines of Bash without modifying core Tilt! Here it is!
```

Comment:
Something like this might be appealing, but feel free to ignore:
"By exposing its internals via an API, Tilt makes easy to add/prototype features - you don't have to worry about getting code into main before you can use it, or breaking existing code, and you can write in any language you want. In fact, you can even use bash!"

Reply:
done